cmd/state-svc/internal/rtwatcher/watcher.go

```diff
@@ -148,6 +140,21 @@ func (w *Watcher) Watch(pid int, exec, source string, dims *dimensions.Values) {
 	logging.Debug("Watching %s (%d)", exec, pid)
 	dims.Sequence = ptr.To(-1) // sequence is meaningless for heartbeat events
 	e := entry{pid, exec, source, dims}
+	w.mutex.Lock()
 	w.watching = append(w.watching, e)
+	w.mutex.Unlock()
 	go w.RecordUsage(e) // initial event
 }
+
+func (w *Watcher) Unwatch(execPath string) {
+	logging.Debug("Unwatching %s", execPath)
+	w.mutex.Lock()
+	defer w.mutex.Unlock()
+	for i, proc := range w.watching {
+		if execPath != proc.Exec {
+			continue
+		}
+		w.watching = append(w.watching[:i], w.watching[i+1:]...)
+		break
+	}
+}
```

Comment:
Please drop this, to handle this properly we would also need to fire the appropriate analytical event which is too much added scope here imo.

Reply:
In my testing, I invoked `python` from a `state shell`. Upon exiting Python, the exit code event triggered an unwatching of that executable, so I was immediately able to run `state install` without waiting for 1 minute. I only had to add this little bit of plumbing. Are there other use cases I'm not thinking of?

We can take this to slack or stand tomorrow. Given the fact that process checking on the fly will always result in a noticible hiccup on macOS, what we want to do may need a more general refactor/rearchitecture/reimplementation.